src/azul/openapi.py

```diff
@@ -125,25 +142,70 @@ def clean_specs(specs):
 
 
 def get_app_specs(app):
+    """
+    Extract OpenAPI specs from a Chalice app object
+    """
     path_specs = {}
     method_specs = {}
     for entries in app.routes.values():
         func = next(iter(entries.values())).view_function
-        if hasattr(func, 'path_specs'):
+        try:
```

Comment:
Each `.` in the body of this try could potentially raise an AttributeError. The body of the `try` should be minimal (one `.` or one use of `[]` for dicts) for so as to not mask/conflate other issues.

Also note that it IS ok to use `dict.get()` and `getattr()` with a default since they use EAFP under the hood.

This might seem pedantic, but this is worse now than the previous version. I try to stick with EAFP as habit so I don't need to think about when it does matter (performance-critical or concurrent code) and when it doesn't.

Reply:
I agree this is worse. Good point about `getattr()`. I've switched to that since I think it is the most readable, yet still EAFP.